Automattic/newspack-network
# feat(content-distribution): sync post meta

Implements support for syncing post meta when distributing content. The new `Newspack_Network\Content_Distribution::get_reserved_post_meta_keys()` should determine which keys to ignore on distribution, to be used by both `Outgoing_Post` when generating the payload and `Incoming_Post` when clearing the post meta before an update.

This approach supports unique and not unique post meta.

### Testing

1. Make sure you have a network setup with at least one hub and one node
2. Setup content distribution by creating a new post and running in wp shell:

```php
$post_id = 0; // Change to your post ID.
$outgoing_post = new Newspack_Network\Content_Distribution\Outgoing_Post( $post_id );
$outgoing_post->set_config( [ 'http://node1.local' ] ); // Change to match your node URL.
Newspack_Network\Content_Distribution::distribute_post( $outgoing_post );
```

3. Still in wp shell, update the post meta and distribute again:

```php
$post = $outgoing_post->get_post();
update_post_meta( $post->ID, 'single', 'value' ); // Single string
update_post_meta( $post->ID, 'array', [ 1, 2, 3 ] ); // Single array
add_post_meta( $post->ID, 'multiple', 'a' ); // Not unique meta
add_post_meta( $post->ID, 'multiple', 'b' ); // Not unique meta
Newspack_Network\Content_Distribution::distribute_post( $outgoing_post );
```

4. Once the events are processed, confirm the meta is updated correctly in the node:

```php
get_post_meta( $post_id );
// = [
//      "single" => [ "value" ],
//      "array" => [ "a:3:{i:0;i:1;i:1;i:2;i:2;i:3;}" ],
//      "multiple" => [ "a", "b", "c" ]
// ]
```
